### PR TITLE
Fix incorrect tuner error message

### LINKED ISSUE #15855: Incorrect error message: Trainer is already configured with a `BatchSizeFinder` callback
### Bug description

I have a trainer with a `LearningRateFinder` callback. In trying to get it to work, I've set `auto_lr_find=True,` on the trainer and have called `trainer.tune()` (I don't know what I'm doing, as you can probably tell).

The bug is that the error message says "`BatchSizeFinder` callback", when it should say "`LearningRateFinder` callback".

Offending code is here: https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/tuner/tuning.py#L299-L303 as at version 1.8.3.post1.

BTW, I'm finding the docs around `auto_lr_find`, `LearningRateFinder`, `trainer.tune()`, `trainer.tuner.lr_find()` and when/where I can set `max_lr` and `min_lr`, and which methods update the model and which don't all quite confusing. Also `_LRFinder` is internal, so not documented (on the website) but returned to the user so probably should be public and exposed as `LrFinderResults` or something?

### How to reproduce the bug

_No response_

### Error messages and logs

```

# Error messages and logs here please

```

### Environment

_No response_

### More info

_No response_

cc @borda

### PR DESCRIPTION
**Pull request by Al3xDo**
## What does this PR do?

Fixes #15855 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/tuner/tuning.py [hunk 1]
```diff
@@ -296,8 +296,19 @@ def _check_tuner_configuration(
...
+    if len(configured_cb) == 1:
         raise MisconfigurationException(
-            "Trainer is already configured with a `BatchSizeFinder` callback. Please remove it if you"
+            f"Trainer is already configured with a `{configured_cb[0]}` callback. Please remove it if you"
             " want to use tuner."
         )
+    elif len(configured_cb) == 2:
+        raise MisconfigurationException(
+            f"Trainer is already configured with `LearningRateFinder` and `BatchSizeFinder` callbacks."
+            " Please remove it if you want to use tuner."
```
Comment (awaelchli) [inline src/pytorch_lightning/tuner/tuning.py:313]: While we are at it, let's make it a full sentence (also above):
```suggestion
            f"Trainer is already configured with `LearningRateFinder` and `BatchSizeFinder` callbacks."
            " Please remove it if you want to use the Tuner."
```

Comment (awaelchli) [reply src/pytorch_lightning/tuner/tuning.py:313]: Also, the "please remove it" should probably become "please remove them" in the case of multiple callbacks.

File: src/pytorch_lightning/tuner/tuning.py [hunk 2]
```diff
@@ -296,8 +296,19 @@ def _check_tuner_configuration(
                 " arguments should be None, please consider setting `dataloaders` instead."
             )
 
-    if any(isinstance(cb, (BatchSizeFinder, LearningRateFinder)) for cb in trainer.callbacks):
+    configured_cb = list()
```
Comment (awaelchli) [inline src/pytorch_lightning/tuner/tuning.py:299]:
```suggestion
    configured_callbacks = []
```

File: tests/tests_pytorch/tuner/test_tuning.py [hunk 3]
```diff
@@ -25,3 +26,37 @@ def test_tuner_with_distributed_strategies():
 
     with pytest.raises(MisconfigurationException, match=r"not supported with `Trainer\(strategy='ddp'\)`"):
         trainer.tune(model)
+
+
+def test_tuner_with_already_configured_BatchSizeFinder():
```
Comment (awaelchli) [inline tests/tests_pytorch/tuner/test_tuning.py:31]: Nit: could you avoid the upper case characters in the function name? :)

**Review by awaelchli — APPROVED** [review]
LGTM, minor comments

**Comment by awaelchli** [comment]
FYI, the existing test `test_if_lr_finder_callback_already_configured` needs to be adjusted because of the changed logic.

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/tuner/tuning.py [hunk 4]
```diff
@@ -296,8 +296,19 @@ def _check_tuner_configuration(
...
+    if len(configured_callbacks) == 1:
         raise MisconfigurationException(
-            "Trainer is already configured with a `BatchSizeFinder` callback. Please remove it if you"
-            " want to use tuner."
+            f"Trainer is already configured with a `{configured_callbacks[0]}` callback."
+            "Please remove it if you want to use the Tuner."
+        )
+    elif len(configured_callbacks) == 2:
+        raise MisconfigurationException(
+            f"Trainer is already configured with `LearningRateFinder` and `BatchSizeFinder` callbacks."
```
Comment (Borda) [inline src/pytorch_lightning/tuner/tuning.py:312]:
```suggestion
            "Trainer is already configured with `LearningRateFinder` and `BatchSizeFinder` callbacks."
```

**Review by awaelchli — APPROVED** [review]

**Pull request merged**